### ISSUE TITLE
Error loading MAML help file

### ISSUE THREAD

**Issue by pkocian** [SitecorePowerShell/Console#317]
After installation of version 3.0 on Sitecore 7.2 Update-1 there is an error loading file Cognifide.PowerShell.dll-Help.xml in the log.
The file has no byte order mark and the XML header declares UTF-16 encoding. I can get rid of this error by changing the XML encoding to UTF-8 or converting the file to UCS-2 Big Endian encoding.
I am not sure about the correct solution, powershell seems to be working fine. In version 2.8 we had no such error. 
Could you replicate this issue or is it just in our environment? 

Here is the trace:
14:17:49,997 [  272] ERROR global Sitecore.Web.UI.XmlControls.FolderControlSource - Failed to load XML control: c:\projects\online\main\wwwroot\sitecore modules\PowerShell\Assets\Cognifide.PowerShell.dll-Help.xml
System.Xml.XmlException: There is no Unicode byte order mark. Cannot switch to Unicode.
   at System.Xml.XmlTextReaderImpl.Throw(Exception e)
   at System.Xml.XmlTextReaderImpl.CheckEncoding(String newEncodingName)
   at System.Xml.XmlTextReaderImpl.ParseXmlDeclaration(Boolean isTextDecl)
   at System.Xml.XmlTextReaderImpl.Read()
   at System.Xml.XmlReader.MoveToContent()
   at Sitecore.Web.UI.XmlControls.FolderControlSource.IsXmlControl(String controlFile)

**Comment by michaellwest** [comment]
I will look into the code used to generate the file. Maybe we need to
specify utf-8 without BOM.

**Comment by michaellwest** [comment]
Looks like this line in the help generator can be updated to write with the
UTF-8 encoding.

File.WriteAllText(Path.Combine(outputPath, string.Format("{0}.dll-help.xml",
asm.GetName().Name)),sb.ToString());

**Comment by michaellwest** [comment]
@pkocian Can you extract the file from the package manually and determine the encoding? I'm seeing UTF-8 encoding for the xml by default.

**Comment by pkocian** [comment]
Yes, I'm seeing the encoding as UTF-8 w/o BOM too. The error disappears only when I change the xml header to <?xml version="1.0" encoding="utf-8"?> or when I convert the file to UCS-2. It is strange that with previous version there was no error even though the encoding of this file was the same.

**Comment by michaellwest** [comment]
The light bulb just went off in my head. I now understand what is happening. Essentially Sitecore is trying to load our PowerShell help file as if it were one of the Sheer UI control files. It's not an issue with the encoding of the file.

**Comment by michaellwest** [comment]
The web.config by default contains this entry for controlSources which essentially includes our help file. 

```powershell
    <controlSources>
...
      <source mode="on" namespace="Sitecore.Web.UI.XmlControls" folder="/sitecore modules" deep="true"/>
...
    </controlSources>
```

@AdamNaj any ideas on how to exclude the directory?

**Comment by AdamNaj** [comment]
the file was renamed to have .maml extension and will no longer be picked up by Sitecore as a control file.

**Comment by AdamNaj** [comment]
Merged by implementing Issue #350
